src/main.rs

```diff
@@ -46,26 +47,32 @@ enum Command {
     },
 }
 
-#[tokio::main]
-async fn main() -> Result<()> {
+fn main() -> Result<()> {
     tracing_subscriber::fmt::init();
 
-    let args = Args::parse();
-    match args.command {
-        Command::Local {
-            local_host,
-            local_port,
-            to,
-            port,
-            secret,
-        } => {
-            let client = Client::new(&local_host, local_port, &to, port, secret.as_deref()).await?;
-            client.listen().await?;
-        }
-        Command::Server { min_port, secret } => {
-            Server::new(min_port, secret.as_deref()).listen().await?;
-        }
-    }
+    let Args { command } = Args::try_parse()?;
```

Comment:
Ah, sure thing! I just figured that since we're using `anyhow`, it would be wise to forward the errors anyway. But you are right that Clap does some special things when parsing fails. I shall revert this in the next commit.